### PR TITLE
Feat: New composite action to inject an issue-id into commit msg

### PR DESCRIPTION
**Pull request by ModeSevenIndustrialSolutions**
Also provides micro-actions for perfoming a key/value lookup in a JSON variable. Another micro-action to fetch the message body, and one more to configure GIT operations in GitHub, which conveniently defaults to the GHA automation/BOT user account.

Tested here:
https://github.com/ModeSevenIndustrialSolutions/test-issue-id/actions/runs/12011503337
https://github.com/ModeSevenIndustrialSolutions/test-issue-id/actions/runs/12011503337/job/33480638748

### CODE REVIEW

**Comment by askb** [comment]
Thanks for contributing this @ModeSevenIndustrialSolutions. I was thinking of a much simpler solution for the dependabot integration. 

Has this been tested to work with the dependabot?  Please add more documentation Readme on how this fits in and the purpose of the micro action!

File: .github/actions/inject-issue-id-action/action.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ github.head_ref }}
+        fetch-depth: 0
+
+    - name: "Get GIT commit message body"
+      id: commit-message
+      # yamllint disable rule:line-length
+      uses: ModeSevenIndustrialSolutions/issue-id/.github/actions/git-commit-message-action@main
```
Comment (tykeal) [inline .github/actions/inject-issue-id-action/action.yaml:37]: This is not pointing to the micro-action that is part of the PR

File: .github/actions/inject-issue-id-action/action.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      run: |
+        # Set key to use for JSON lookup
+        ACTOR="${{ github.actor }}"
+        ACTOR_ID="${{ github.actor_id }}"
+        echo "Using GitHub actor as lookup key: $ACTOR [$ACTOR_ID] 🔑"
+        echo "key=$ACTOR" >> "$GITHUB_ENV"
+
+    - name: "Get ticket from JSON lookup table"
+      if: env.PRESENT == 'false' && inputs.INJECT == 'true'
+      uses: ModeSevenIndustrialSolutions/issue-id/.github/actions/json-key-value-lookup-action@main
```
Comment (tykeal) [inline .github/actions/inject-issue-id-action/action.yaml:71]: This is not pointing to the micro-action that is part of the PR

File: .github/actions/inject-issue-id-action/action.yaml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    value: ${{ steps.status.outputs.present }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ github.head_ref }}
+        fetch-depth: 0
```
Comment (tykeal) [inline .github/actions/inject-issue-id-action/action.yaml:32]: Why are you forcing a complete checkout? Given that this is expected to only be used on PRs and not Gerrit changes, this should really be using the default of 1 for faster checkouts.

File: .github/actions/inject-issue-id-action/action.yaml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    - name: "Get ticket from JSON lookup table"
+      if: env.PRESENT == 'false' && inputs.INJECT == 'true'
+      uses: ModeSevenIndustrialSolutions/issue-id/.github/actions/json-key-value-lookup-action@main
+      with:
+        json: ${{ inputs.issue_id_lookup_json }}
+        key: ${{ env.key }}
+
+    - name: "Configure GIT user for commit"
+      if: env.PRESENT == 'false' && inputs.INJECT == 'true'
+      uses: ModeSevenIndustrialSolutions/issue-id/.github/actions/git-configure-action@main
```
Comment (tykeal) [inline .github/actions/inject-issue-id-action/action.yaml:78]: This is not pointing to the micro-action that is part of the PR

**Review by tykeal — CHANGES_REQUESTED** [review]
Please add documentation.

Ideally each of these micro-actions would be a separate commit, with documentation.

The `inject-issue-id-action` is still pointing to the micro-actions in your `issue-id` repository and not referencing the micro-actions that you are adding

File: .github/actions/inject-issue-id-action/action.yaml (hunk 1 above)
Comment (ModeSevenIndustrialSolutions) [reply .github/actions/inject-issue-id-action/action.yaml]: Why is it I always forget to update these self-references when submitting patches?

File: .github/actions/inject-issue-id-action/action.yaml (hunk 3 above)
Comment (ModeSevenIndustrialSolutions) [reply .github/actions/inject-issue-id-action/action.yaml]: I was trying to address the issues around the force push and I think I misunderstood what was happening during the checkout step. "github.head_ref" was the required fix and the fetch depth wasn't required. I've updated the checkout step accordingly, and tested again today in the workflow run below:
https://github.com/ModeSevenIndustrialSolutions/test-issue-id/actions/runs/12120595981/job/33789781183

**Comment by ModeSevenIndustrialSolutions** [comment]
I will address the missing documentation this week, as requested.

**Comment by ModeSevenIndustrialSolutions** [comment]
I have added the missing documentation, but as per your recommendation, will refactor this effort into four separate pull requests.

**Comment by ModeSevenIndustrialSolutions** [comment]
Moved to https://github.com/lfit/releng-reusable-workflows/pull/202

**Pull request closed without merging**